docs/guides/transactions/running-background-jobs.md

```diff
@@ -45,25 +38,19 @@ namespace StarcounterApplication4
 
 ## Basic Information About Scheduling
 
-Each Starcounter scheduler has a queue of tasks that are supposed to be run on this scheduler. Tasks are picked from the queue and executed. To put a task in a queue, the `Scheduling.RunTask` should be used. When scheduling a task, you can specify the scheduler number, and if the thread should wait for the task to be picked by scheduler and completed. Here is the signature of the `Scheduling.RunTask`:
+Tasks scheduled by `Scheduling.RunTask` as well as ready-to-call handlers make a queue of ready tasks. Then Starcounter picks tasks and runs them on a free thread from the thread pool. By default Starcounter thread pool hosts number of threads equal to number of CPU cores. It allows to maximize throughput by avoiding unnecessary context switches. But what if user code happens to block on IO or synchronization? Starcounter is able to detect such condition and launch an additional thread, so that queue processing is not stall. To maximize throughput it's not recommended to rely on thread pool scaling. Better to avoid blocking calls and use async IO whenever possible. 
```

Comment:
"ready-to-call handlers"  is not clear to me. Do you think you can rephrase? As I understand it, these are, for example, the methods of the `Handle` API.

 "processing is not stall" -> "does not stall"

"To maximize throughput it's not recommended to rely on" -> "To maximize throughput, don't rely on"

Reply:
>"ready-to-call handlers" is not clear to me. Do you think you can rephrase? As I understand it, these are, for example, the methods of the Handle API.

These are handlers that are scheduled in the queue because we have some data to handle.


>"processing is not stall" -> "does not stall"
>"To maximize throughput it's not recommended to rely on" -> "To maximize throughput, don't rely on"

will fix